### ISSUE TITLE
ERR_INVALID_OPT_VALUE when using create-single-spa

### ISSUE THREAD

**Issue by dcalise** [single-spa/create-single-spa#100]
```
$ create-single-spa
(node:59293) ExperimentalWarning: The fs.promises API is experimental
? Directory for new project .
? Select type to generate single-spa root config
? Which package manager do you want to use? npm
? Will this project use Typescript? No
events.js:167
      throw er; // Unhandled 'error' event
      ^

TypeError [ERR_INVALID_OPT_VALUE]: The value "undefined" is invalid for option "flags"
    at stringToFlags (internal/fs/utils.js:253:9)
    at open (internal/fs/promises.js:198:34)
    at Object.readFile (internal/fs/promises.js:466:20)
    at SingleSpaRootConfigGenerator.createPackageJson (/Users/dcalise/.nvm/versions/node/v10.2.1/lib/node_modules/create-single-spa/node_modules/generator-single-spa/src/root-config/generator-root-config.js:49:42)
    at process._tickCallback (internal/process/next_tick.js:68:7)
Emitted 'error' event at:
    at Immediate.setImmediate (/Users/dcalise/.nvm/versions/node/v10.2.1/lib/node_modules/create-single-spa/node_modules/yeoman-generator/lib/index.js:791:20)
    at runCallback (timers.js:696:18)
    at tryOnImmediate (timers.js:667:5)
    at processImmediate (timers.js:649:5)
```

I've tried both via `npx create-single-spa` as well is an `npm -g` install and `create-single-spa` command.

**Comment by joeldenning** [comment]
Please upgrade your version of nodejs to 10.4.0 or later. I see you're on 10.2.1, which had a bug that was fixed that create-single-spa relies on.

I'm closing this issue and will add the minimum nodejs requirements to the documentation.

**Comment by joeldenning** [comment]
I decided to add a check to create-single-spa instead of just adding it to the documentation. #101 adds a minimum nodejs check to the start of create-single-spa so that this will be surfaced better for those in the future

**Comment by joeldenning** [comment]
The check for NodeJS version is implemented in 1.7.0
